addons/stock_account/models/account_move.py

```diff
@@ -315,3 +315,13 @@ def _stock_account_get_anglo_saxon_price_unit(self):
     @api.onchange('product_id')
     def _inverse_product_id(self):
         super(AccountMoveLine, self.filtered(lambda l: l.display_type != 'cogs'))._inverse_product_id()
+
+    def _get_exchange_journal(self, company):
+        if self and self.product_id.categ_id.property_valuation == 'real_time':
```

Comment:
seems good but I would still check there is a valuation layer linked to the account move.